src/adaptation/icap/ModXact.cc

```
@@ -1157,6 +1157,31 @@ void Adaptation::Icap::ModXact::decideOnParsingBody()
     }
 }
 
+namespace Adaptation {
+namespace Icap {
+class ExtensionsParser : public Http1::CustomExtensionsParser
+{
+public:
+    ExtensionsParser() : useOriginBody_(-1) {}
+
+    virtual bool parse(Tokenizer &tok, const SBuf &extName) override {
+        assert(knownExtension(extName));
+        return parseIntExtension(tok, UseOriginalBodyName, useOriginBody_);
```

We should throw an error on negative values, right?